runtime/lib/Runtime/index.ts

```diff
@@ -186,24 +186,24 @@ class Runtime<
     await super.callEvent<K>(type, event, this);
   }
 
-  public getProgram(programID: string) {
-    return this.programManager.get(programID);
+  public getProgram(versionID: string, diagramID: string) {
+    return this.programManager.get(versionID, diagramID);
   }
 
   private async injectBaseProgram() {
-    if (this.stack.get(0)?.getProgramID() === this.versionID) {
+    if (this.stack.get(0)?.getDiagramID() === this.versionID) {
       return;
     }
 
     // insert base program to the stack
-    const program = await this.api.getProgram(this.versionID).catch(() => null);
+    const program = await this.api.getProgram(this.versionID, this.versionID).catch(() => null);
     if (!program) {
       return;
     }
 
     this.stack.unshift(
       new Frame({
-        programID: this.versionID,
+        diagramID: this.versionID,
```

Comment:
so we are storing `versionID` into a `diagramID`, probably it makes sense to have a different name in the `Frame` just to show not only `diagramID` is stored in this field...

```suggestion
        programID: this.versionID,
        // or
        frameID: this.versionID,
```

Reply:
No sure if I get the idea.

In this case, the `diagramID` will be the `versionID` because it's the base program. 
During the compilation we create a root program, which its `_id` and `diagramID` will be the same as the `versionID`.

That's why I think it still makes sense to keep the diagramID property.